website/docs/migration/from_state_notifier.mdx

```diff
@@ -0,0 +1,386 @@
+---
+title: From `StateNotifier` and `ChangeNotifier`
+---
+
+import buildIinit from "./build_init";
+import familyAndDispose from "./family_and_dispose";
+import asyncNotifier from "./async_notifier";
+import addListener from "./add_listener";
+import fromChangeNotifier from "./from_change_notifier";
+import fromStateProvider from "./from_state_provider";
+import oldLifecycles from "./old_lifecycles";
+import { AutoSnippet } from "../../src/components/CodeSnippet";
+
+Riverpod went `2.0`! To see the full list of changes, consult the
+[Changelog](https://pub.dev/packages/flutter_riverpod/changelog#200).
+
+Riverpod `2.0` introduced the new `Notifier` and `AsyncNotifier` classes.  
+This page shows how to migrate from the deprecated `StateNotifier` (and from the legacy `ChangeNotifier`)
+to the new APIs.
+
+The main benefit introduced by `AsyncNotifier` is a better `async` support; indeed,
+`AsyncNotifier` can be thought as a `FutureProvider` with methods.
+
+Furthermore, the new `Notifier`s:
+
+- Expose a `Ref` object inside its class
+- Offer similar syntax between codegen and non-codegen approaches
+- Offer similar syntax between their sync and async versions
+- Move away logic from Providers and centralize it into the Notifiers themselves
+
+Let's see how to define a `Notifier`, how it compares with `StateNotifier` and explore
+the new `AsyncNotifier` for asynchronous state classes.
+
+## The new syntax
+
+Let's write some simple logic with a synchronous `Notifier`:
+
+<AutoSnippet language="dart" {...buildIinit}></AutoSnippet>
+
+Notice the following design choices:
+
+- Notifiers are initialized in their `build` method, along with their reactive dependencies
+- Notifiers expose `ref` into their class, whereas their Provider don't
+
+### Comparison with `StateNotifier`
+
+Let's compare the new APIs with the old ones; here, we implement the above example with `StateNotifier`:
+
+```dart
```

Comment:
No code blocks in docs please :)

Reply:
You don't need to specify hooks/codegen objects. Only the "raw" parameter is required